cooked-validators/src/Cooked/Tx/Constraints/Type.hs

```diff
@@ -214,10 +214,31 @@ data TxOpts = TxOpts
     --
     -- /This has NO effect when running in 'Plutus.Contract.Contract'/.
     -- By default, this is set to @True@.
-    balance :: Bool
+    balance :: Bool,
+
+    -- | The 'AdjustOutputPolicy' to apply when balancing the transaction.
+    --
+    -- /This has NO effect when running in 'Plutus.Contract.Contract'/.
+    -- By default, this is set to @AdjustExistingOutput@.
+    adjustOutputPolicy :: AdjustOutputPolicy
+
   }
   deriving (Eq, Show)
 
+
+{-| Whether to adjust existing public key outputs during
+transaction balancing.
+-}
+data AdjustOutputPolicy
+  = AdjustExistingOutput
+  -- ^ Try to adjust an existing public key output with the change. If no
+  --   suitable output can be found, create a new change output.
+  | DontAdjustExistingOutput
+  -- ^ Do not change the existing outputs, always create a new change
+  --   output.
+  deriving (Eq, Ord, Show)
+
+
 -- IMPORTANT INTERNAL: If you add or remove fields from 'TxOpts', make sure
 -- to update the internal @fields@ value from 'Cooked.Tx.Constraints.Pretty'
```

Comment:
This also needs to be done in order to keep the pretty printer consistent; all that is necessary is to add the following code to [this list](https://github.com/tweag/plutus-libs/blob/0a8b5f8d80ea446c03c79601910aac3796814199/cooked-validators/src/Cooked/Tx/Constraints/Pretty.hs#L116)

```patch
        Field "unsafeModTx" unsafeModTx,
+       Field "balanceOutputPolicy" balanceOutputPolicy,
        Field "balance" balance
      ]
```

(assuming that "adjust" gets replaced by "balance" in the respective names :) )

Reply:
Pretty printer updated